pkg/apis/portworx/portworx.go

```diff
@@ -0,0 +1,9 @@
+// Package v1 contains API Schema definitions for the portworx API group
+// +k8s:deepcopy-gen=package,register
+// +groupName=portworx.io
+package portworx
```

Comment:
operator was designed for any storage drivers before right? then we're adding `pkg/apis/portworx`, moving forward are we still considering that when adding new components? because this path doesn't seem to be generic.

Reply:
Yes, it was meant to be generic. That is why we created a new API group for the portworx APIs like the Diag so if needed we can move them to a different git repo/container and does not affect any of the 'operator' APIs. To avoid creating multiple containers we are putting the diags controller in the portworx operator. But again, both the controller and API can be moved out easily if needed.

Moving forward we should follow the same guideline. If it is very generic and belongs to the `core` operator functionality then it can go under core. For portworx specific CRs we should use the `portworx.io` group.